pkg/autoscaler/statforwarder/processor.go

```diff
@@ -39,23 +39,30 @@ type bucketProcessor struct {
 	accept statProcessor
 }
 
-func (p *bucketProcessor) process(sm asmetrics.StatMessage) {
+func (p *bucketProcessor) process(sm asmetrics.StatMessage) error {
 	l := p.logger.With(zap.String("revision", sm.Key.String()))
 	if p.accept != nil {
 		l.Debug("Accept stat as owner of bucket ", p.bkt)
 		p.accept(sm)
-		return
+		return nil
 	}
 
 	l.Debugf("Forward stat of bucket %s to the holder %s", p.bkt, p.holder)
 	wsms := asmetrics.ToWireStatMessages([]asmetrics.StatMessage{sm})
 	b, err := wsms.Marshal()
 	if err != nil {
-		l.Errorw("Error while marshaling stats", zap.Error(err))
-		return
+		return err
 	}
 
 	if err := p.conn.SendRaw(gorillawebsocket.BinaryMessage, b); err != nil {
-		l.Errorw("Error while sending stats", zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+func (p *bucketProcessor) shutdown() {
+	if p.conn != nil {
```

Comment:
mostly out of interest, Im trying to figure out if and how `p.conn` can actually be nil? Is there a path where that actually happens or is this just caution?

Reply:
If `accept` is not nil (the pod is the owner of the bucket and it doesn't need to forward), `conn` is nil.